src/CodeGen/GenerateSyntax.hs

```diff
@@ -80,9 +83,17 @@ toType xs = foldr1 combine $ map convertToQType xs
 toCamelCase :: String -> String
 toCamelCase = initUpper . mapOperator . removeUnderscore
 
+clashingNames :: HashSet String
+clashingNames = HashSet.fromList ["type"]
```

Comment:
I’m pretty sure we’ll get better performance with a `case`ing predicate given the relatively small set of reserved words we’re expecting:

```haskell
isReserved :: String -> Bool
isReserved "type" = True
isReserved _      = False

addTickIfNecessary :: String -> String
addTickIfNecessary s
  | isReserved s = s ++ "'"
  | otherwise    = s
```